src/Core/Validation/FieldValidation/YearFieldValidator.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace SilverStripe\Core\Validation\FieldValidation;
+
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\ORM\FieldType\DBYear;
+
+/**
+ * Validates a value in DBYear field
+ * This FieldValidator is not intended to be used in fields other than DBYear
+ */
+class YearFieldValidator extends IntFieldValidator
+{
+    private const YEAR_EXCEPTION = 0;
```

Comment:
What is "YEAR_EXCEPTION"? It sounds like an `Exception` but it's clearly not that. Please add a PHPDoc.

Reply:
Yes bad naming.  Renamed to SPECIAL_YEAR and added docblock